src/framework/theme/components/chat/chat-message.component.ts

```diff
@@ -193,15 +202,34 @@ export class NbChatMessageComponent {
    */
   @Input() type: string;
 
+  @Input() customMessageData: any;
+
+  private readonly defaultMessageTypes: string[] = ['text', 'file', 'map', 'quote', undefined];
+
   constructor(protected domSanitizer: DomSanitizer) { }
 
   getInitials(): string {
     if (this.sender) {
       const names = this.sender.split(' ');
-
       return names.map(n => n.charAt(0)).splice(0, 2).join('').toUpperCase();
     }
-
     return '';
   }
+
+  isDefaultMessageType(msgType: string): boolean {
+    return this.defaultMessageTypes.some(el => el === msgType);
+  }
+
+  getActualTemplate(type: any): any {
+    if (!type) {
+      throw new Error('Custom template type must be provided');
+    }
+
+    const customMessage = this.customMessages.find(msg => msg.customMessageType === type);
+    if (customMessage === undefined || !customMessage.templateRef) {
+      throw new Error(`Can't find template for custom message type = ${type}.`);
```

Comment:
Change message to `nb-chat: Can't find template for custom type "${type}". Make sure you provide it in the chat component with "*nbCustomMessage="'${type}'".`